src/minio/managers/policy_manager.py

```diff
@@ -135,54 +155,117 @@ async def create_group_policy(self, group_name: str) -> PolicyModel:
-            resource_paths.append(
-                f"{self.config.groups_general_warehouse_prefix}/{target_name}"
-            )
+            return [
+                f"{self.config.users_sql_warehouse_prefix}/{target_name}",
+                f"{self.config.users_general_warehouse_prefix}/{target_name}",
+            ]
+        elif target_type == TargetType.GROUP:
+            return [f"{self.config.groups_general_warehouse_prefix}/{target_name}"]
+
+    def _create_policy_statements(
+        self, resource_paths: list[str]
+    ) -> list[PolicyStatement]:
+        """Create all required policy statements for S3 access."""
+        return [
+            self._create_list_all_buckets_statement(),
+            self._create_bucket_location_statement(),
+            self._create_list_bucket_statement(resource_paths),
+            self._create_object_operations_statement(resource_paths),
+        ]
+
+    def _create_list_all_buckets_statement(self) -> PolicyStatement:
+        """Create statement for listing all buckets (required for MinIO users to see buckets in the UI)."""
+        return PolicyStatement(
+            effect=PolicyEffect.ALLOW,
+            action=[PolicyAction.LIST_ALL_MY_BUCKETS],
+            resource=["*"],
+            condition=None,
+            principal=None,
+        )
 
-        # Build resource ARNs from paths
-        resources = []
+    def _create_bucket_location_statement(self) -> PolicyStatement:
+        """Create statement for getting bucket location (required for MinIO users to see buckets in the UI)."""
+        return PolicyStatement(
+            effect=PolicyEffect.ALLOW,
+            action=[PolicyAction.GET_BUCKET_LOCATION],
+            resource=[f"arn:aws:s3:::{self.config.default_bucket}"],
+            condition=None,
+            principal=None,
+        )
+
+    def _create_list_bucket_statement(
+        self, resource_paths: list[str]
+    ) -> PolicyStatement:
+        """Create statement for listing bucket contents with path restrictions."""
+        prefix_conditions = []
         for path in resource_paths:
-            resources.append(f"arn:aws:s3:::{self.config.default_bucket}/{path}/*")
-
-        policy_doc = PolicyDocument(
-            statement=[
-                PolicyStatement(
-                    effect=PolicyEffect.ALLOW,
-                    action=[
-                        PolicyAction.GET_OBJECT,
-                        PolicyAction.PUT_OBJECT,
-                        PolicyAction.DELETE_OBJECT,
-                        PolicyAction.LIST_BUCKET,
-                    ],
-                    resource=resources,
-                    condition=None,
-                    principal=None,
-                )
-            ]
+            prefix_conditions.extend([f"{path}/*", f"{path}"])
+
+        return PolicyStatement(
+            effect=PolicyEffect.ALLOW,
+            action=[PolicyAction.LIST_BUCKET],
+            resource=[f"arn:aws:s3:::{self.config.default_bucket}"],
```

Comment:
Will users only ever have access to a single bucket? What if we have test buckets, special buckets for CTS output, etc.?

Reply:
hmmm that's a good question. We might need it. But for the purpose of data governance - this should be fine at the moment.

Reply:
yea - I think we want to deliver the data governance for now - this service is not primarily designed to handle bucket permissions.

Reply:
It's - but focusing on `sql-warehouse` for now.

Reply:
That's fine for now, yeah. All I'm saying is we're not going to want to have multiple different systems for determining what resources in S3 are accessible to the user

Reply:
yea - I agree.